scripts/build-xcframework.sh

```diff
@@ -19,18 +19,38 @@ generate_xcframework() {
     
     local resolved_configuration="Release$configuration_suffix"
     local resolved_product_name="$scheme$configuration_suffix"
-    
+    local OTHER_LDFLAGS="-Wl,-make_mergeable"
+
     if [ "$MACH_O_TYPE" = "staticlib" ]; then
         #For static framework we disabled symbols because they are not distributed in the framework causing warnings.
         GCC_GENERATE_DEBUGGING_SYMBOLS="NO"
+        OTHER_LDFLAGS="" # Disable mergeable libraries for static libs
     fi
     
     rm -rf Carthage/DerivedData
     
     for sdk in "${sdks[@]}"; do
         if grep -q "${sdk}" <<< "$ALL_SDKS"; then
 
-            xcodebuild archive -project Sentry.xcodeproj/ -scheme "$scheme" -configuration "$resolved_configuration" -sdk "$sdk" -archivePath "./Carthage/archive/${scheme}${suffix}/${sdk}.xcarchive" CODE_SIGNING_REQUIRED=NO SKIP_INSTALL=NO CODE_SIGN_IDENTITY= CARTHAGE=YES MACH_O_TYPE="$MACH_O_TYPE" ENABLE_CODE_COVERAGE=NO GCC_GENERATE_DEBUGGING_SYMBOLS="$GCC_GENERATE_DEBUGGING_SYMBOLS"
+            ## watchos and watchsimulator dont support make_mergeable: ld: unknown option: -make_mergeable
+            if [[ "$sdk" == "watchos" || "$sdk" == "watchsimulator" ]]; then
+                OTHER_LDFLAGS=""
+            fi
```

Comment:
h: OTHER_LDFLAGS gets cleaned here and is not set again. visionOS and macCatalyst don’t get the OTHER_LDFLAGS.